src/authenticator/formAuthenticator.js

```diff
@@ -121,31 +115,31 @@ FormAuthenticator.prototype.get = function(url){
   return browser.testrunner.navigation.waitForRedirect(this.redirectUrl || url);
 };
 
-FormAuthenticator.prototype._wait = function (fnCondition, sTimeoutMessage) {
-  return browser.driver.wait(fnCondition, browser.getPageTimeout, 'Waiting for auth page to fully load. Step: ' + sTimeoutMessage);
+FormAuthenticator.prototype._wait = function (fnCondition, sTimeoutMessage,iTimeout) {
+  return browser.driver.wait(fnCondition, iTimeout || browser.getPageTimeout, sTimeoutMessage);
 };
 
 FormAuthenticator.prototype._waitForField = function (sSelector, sTimeoutMessage) {
   var that = this;
   return this._wait(function () {
-    return that._getFields(sSelector).then(that._checkDisplayed);
+    return that._isPresent(sSelector);
   }, sTimeoutMessage);
 };
 
-FormAuthenticator.prototype._getFields = function (sSelector) {
-  return browser.driver.findElements(by.css(sSelector));
-};
-
 FormAuthenticator.prototype._getField = function (sSelector) {
   return browser.driver.findElement(by.css(sSelector));
 };
 
-FormAuthenticator.prototype._checkDisplayed = function (aFields) {
-  if (aFields.length) {
-    return aFields[0].isDisplayed().then(function (isDisplayed) {
-      return isDisplayed;
-    });
-  }
+FormAuthenticator.prototype._isPresent = function (sSelector) {
```

Comment:
rename to _isEnabled

Reply:
I try to emulate the semantics of isPresent from Protractor and they use isDisplayed and isEnabled there.